src/neo4j_genai/kg_construction/kg_writer.py

```diff
@@ -0,0 +1,144 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+from abc import abstractmethod
+from typing import Optional
+
+import neo4j
+
+from neo4j_genai.core.pipeline import Component
+from neo4j_genai.embedder import Embedder
+from neo4j_genai.exceptions import EmbeddingRequiredError
+from neo4j_genai.indexes import upsert_vector, upsert_vector_on_relationship
+from neo4j_genai.kg_construction.types import Neo4jGraph, Neo4jNode, Neo4jRelationship
+from neo4j_genai.neo4j_queries import UPSERT_NODE_QUERY, UPSERT_RELATIONSHIP_QUERY
+
+
+class KGWriter(Component):
+    """Abstract class used to write a knowledge graph to a data store."""
+
+    @abstractmethod
+    async def run(self, graph: Neo4jGraph) -> None:
```

Comment:
The pipeline currently is not able to instantiate a Pydantic model as component input. However, we can add the `@validate_call` decorator from pydantic, that will do the type cast for us (and raise a ValidationError if it doesn't work)

Reply:
So only base types can be inputs to the `run` function in a `component`?

Reply:
I'm confused, `graph` is still a `Neo4jGraph` in your snippet. Is there an easy way we could change the pipeline to allow for non-base types to be passed as inputs to the `run` function of a `Component`? I think we need a way to ensure the input to this component conforms to a specific graph structure.